File: Ejercicio personajes (Fabrica abstracta)/Fabrica abstracta V2/src/Fabricas/FabricaAbstracta.java
package Fabricas;

import Interfaces.IMontura;
import Interfaces.ICuerpo;
import Interfaces.IArma;
import Interfaces.IArmadura;

public abstract class FabricaAbstracta {
    public abstract IArma crearArma();
    public abstract IArmadura crearArmadura();
    public abstract IMontura crearMontura();
    public abstract ICuerpo crearCuerpo();
}

File: Ejercicio personajes (Fabrica abstracta)/CombatChoose/CombatChoose/fabricas/fabrica_abstracta.py
from abc import ABC, abstractmethod

class FabricaAbstracta(ABC):
    @abstractmethod
    def crear_arma(self):
        pass

    @abstractmethod
    def crear_armadura(self):
        pass

    @abstractmethod
    def crear_cuerpo(self):
        pass

    @abstractmethod
    def crear_montura(self):
        pass
